### PR TITLE
Consume gasRefund From Potential ethereumjs-vm Addition

### PR DESCRIPTION
**Pull request by davidmurdoch**
This is the same as PR #80 with added tests. Fixes issue #26: `estimateGas returns invalid value when tx has a gasRefund`.

### CODE REVIEW

**Comment by seesemichaelj** [comment]
@davidmurdoch FYI, sometimes you can push directly to the PR branch if they checked the box to "allow reviewers to modify it". no problem though!

**Comment by davidmurdoch** [comment]
@seesemichaelj I used Github's "open this in GitHub Desktop" link on the PR and it automatically created a new branch instead of just opening your branch and so I just rolled with it from there. `¯\_(ツ)_/¯`

I'll be sure to do it The Right Way next time.

File: lib/statemanager.js [hunk 1]
```diff
@@ -594,7 +594,7 @@ StateManager.prototype.processGasEstimate = function (from, rawTx, blockNumber,
       }
       var result = '0x0'
       if (!results.error) {
-        result = to.hex(results.gasUsed)
+        result = results.gasRefund ? to.hex(results.gasUsed.add(results.gasRefund)) : to.hex(results.gasUsed);
```
Comment (benjamincburns) [inline lib/statemanager.js]: We expect the `gasRefund` field to always be there, right? If so, this squelches an error and makes things look like they're working if the `gasRefund` field suddenly disappears.

**Comment by benjamincburns** [comment]
It _might_ be worth adding a test to check whether our gas estimation accounts for explicit gas stipends properly (e.g. something like `mycontract.depth.gas(100000)(y-1)`)

Relevant docs [here](http://solidity.readthedocs.io/en/latest/control-structures.html#external-function-calls).

File: lib/statemanager.js (hunk 1 above)
Comment (davidmurdoch) [reply lib/statemanager.js]: fixed in latest commits

**Pull request merged**